src/DataValues/DecimalValue.php

```diff
@@ -360,13 +350,20 @@ public function getArrayValue() {
 	}
 
 	/**
-	 * Constructs a new instance of the DataValue from the provided data.
-	 * This can round-trip with @see getArrayValue
+	 * Constructs a new instance from the provided data. Required for @see DataValueDeserializer.
+	 * This is expected to round-trip with @see getArrayValue.
+	 *
+	 * @deprecated since 0.8.3. Static DataValue::newFromArray constructors like this are
+	 *  underspecified (not in the DataValue interface), and misleadingly named (should be named
+	 *  newFromArrayValue). Instead, use DataValue builder callbacks in @see DataValueDeserializer.
```

Comment:
> Not only was it the only caller of the DataValue::newFromArray methods. The call in DataValueDeserializer was actually the specification of how all DataValue::newFromArray methods must behave.

Wrong. As I wrote, `newFromArray` predates `DataValueDeserializer`, the later being added to migrate away from the former.

> Which is why I started calling this a "circular dependency" in a comment above.

That is poor use of words IMO, since this most certainly is not a circular dependency. You can broaden the term sure, but if you go this far, it becomes so vague that it loses its utility.

> An alternative was added in 2016 with DataValues/Serialization#14.

Yeah cool. This is the kind of thing that was always meant to happen to finish the migration.

---

This is just talking about the past. I'd rather focus on how to proceed now. Anyway, going to ubsub here and leave this in your (collective) capable hands.